drivers/gpio/gpio_rpi_pico.c

```diff
@@ -29,32 +29,52 @@ struct gpio_rpi_data {
 	struct gpio_driver_data common;
 	sys_slist_t callbacks;
 	uint32_t int_enabled_mask;
+	uint32_t single_ended_mask;
+	uint32_t open_drain_mask;
 };
 
 static int gpio_rpi_configure(const struct device *dev,
 				gpio_pin_t pin,
 				gpio_flags_t flags)
 {
-	if (flags & GPIO_SINGLE_ENDED) {
-		return -ENOTSUP;
-	}
+	struct gpio_rpi_data *data = dev->data;
+
+	gpio_set_pulls(pin,
+		(flags & GPIO_PULL_UP) != 0U,
+		(flags & GPIO_PULL_DOWN) != 0U);
 
 	/* Avoid gpio_init, since that also clears previously set direction/high/low */
 	gpio_set_function(pin, GPIO_FUNC_SIO);
 
 	if (flags & GPIO_OUTPUT) {
-		gpio_set_dir(pin, GPIO_OUT);
-
-		if (flags & GPIO_OUTPUT_INIT_HIGH) {
-			gpio_put(pin, 1);
-		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
-			gpio_put(pin, 0);
+		if (flags & GPIO_SINGLE_ENDED) {
+			data->single_ended_mask |= 1 << pin;
+			if (flags & GPIO_LINE_OPEN_DRAIN) {
+				data->open_drain_mask |= 1 << pin;
+				gpio_put(pin, 0);
+			} else {
+				data->open_drain_mask &= ~(1 << pin);
+				gpio_put(pin, 1);
+			}
+			if (((flags & GPIO_LINE_OPEN_DRAIN) &&
```

Comment:
This deserves a short comment explaining what this does - setting the initial state of OE if needed.

Reply:
DONE